### ISSUE TITLE
The comand it fails from start

### ISSUE THREAD

**Issue by catalinux** [nunomaduro/larastan#119]
- Larastan Version: last
- `--level` used: default

### Description:
It fails when I run `php artisan code:analyze`
### Laravel code where the issue was found:

```
$ php artisan code:analyse --paths="app/" --error-format=table
Note: Using the option --errorFormat is deprecated. Use --error-format instead.
Error formatter "" not found. Available error formatters are: checkstyle, raw, table, json, prettyJson
```
When downgrading to phpstan 0.10.2: it works

```
$ composer update
Loading composer repositories with package information
Updating dependencies (including require-dev)
Package operations: 0 installs, 2 updates, 0 removals
  - Downgrading phpstan/phpstan (0.10.3 => 0.10.2): Downloading (100%)
  - Updating psy/psysh (v0.9.6 => v0.9.7): Downloading (100%)

**Comment by ondrejmirtes** [comment]
This is a weird error because in PHPStan itself, there's backwards compatibility to make the old option work:

```
phpstan master $ bin/phpstan analyse src --errorFormat json
Note: Using the option --errorFormat is deprecated. Use --error-format instead.
 336/336 [▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓] 100%

{"totals":{"errors":0,"file_errors":0},"files":[],"errors":[]}
```

**Comment by catalinux** [comment]
in PHPStan they have a wierd check
```
		if ($oldErrorFormat !== null) {
			$errOutput->writeln('Note: Using the option --errorFormat is deprecated. Use --error-format instead.');

			$errorFormat = $oldErrorFormat;
		}
```

The problem is that `$oldErrorFormat`  is empty string.

**Comment by ondrejmirtes** [comment]
It's not weird - you either pass `--error-format` or `--errorFormat`. I don't know why someone would pass `-errorFormat ''`.

**Comment by ondrejmirtes** [comment]
I got it: This line in OptionsResolver is potentially dangerous:

```
$definition = clone $this->command->getDefinition();
```

Because it clones deprecated options as well.

**Comment by catalinux** [comment]
I don't.

I pass

```
$ php artisan code:analyse --paths="app/"
Note: Using the option --errorFormat is deprecated. Use --error-format instead.
Error formatter "" not found. Available error formatters are: checkstyle, raw, table, json, prettyJson
```

The problem is that the oldErrorFormat  gets emptyString and not `null`

**Comment by ondrejmirtes** [comment]
I'm hopefully fixing this in #121.

**Comment by ondrejmirtes** [comment]
Lock PHPStan to 0.10.2 or wait until the PR is merged 😊

**Comment by progalaa** [comment]
does anyone have idea about when the PR will be Merged ? @ondrejmirtes

**Comment by ondrejmirtes** [comment]
Nuno (the maintainer) is on vacation whole August, so probably after that. Use PHPStan 0.10.2 until then.

**Comment by ijpatricio** [comment]
@progalaa @catalinux @ondrejmirtes @coding-sunshine this issue has been fixed on the version v0.2.5, props to @ondrejmirtes
